libzeth/core/group_element_utils.tcc

```diff
@@ -73,21 +74,59 @@ void point_affine_read_json(GroupT &point, std::istream &in_s)
     }
 }
 
-template<typename GroupT> std::string point_affine_to_json(const GroupT &point)
+template<typename GroupT> std::string group_element_to_json(const GroupT &point)
 {
     std::stringstream ss;
-    point_affine_write_json(point, ss);
+    group_element_write_json(point, ss);
     return ss.str();
 }
 
-template<typename GroupT> GroupT point_affine_from_json(const std::string &json)
+template<typename GroupT>
+GroupT group_element_from_json(const std::string &json)
 {
     std::stringstream ss(json);
     GroupT result;
-    point_affine_read_json(result, ss);
+    group_element_read_json(result, ss);
     return result;
 }
 
+template<typename GroupT>
+void group_element_write_bytes(const GroupT &point, std::ostream &out_s)
+{
+    typename std::decay<GroupT>::type affine_p = point;
```

Comment:
Why do you need  the decay type of GroupT here?

Reply:
`GroupT` can sometimes include `const`, which prevents calls to `to_affine_coordinates`.  `std::decay<T>` strips away const, ptr, reference to expose the underlying type.

Reply:
Oh I see, thanks!